wpilibc/src/main/native/include/frc/apriltag/AprilTagFieldLayout.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <string_view>
+#include <vector>
+
+#include <wpi/SymbolExports.h>
+
+#include "frc/DriverStation.h"
+#include "frc/apriltag/AprilTag.h"
+#include "frc/geometry/Pose3d.h"
+
+namespace wpi {
+class json;
+}  // namespace wpi
+
+namespace frc {
+class WPILIB_DLLEXPORT AprilTagFieldLayout {
+ public:
+  AprilTagFieldLayout() = default;
+
+  /**
+   * Construct a new AprilTagFieldLayout with values imported from a JSON file.
+   */
+  explicit AprilTagFieldLayout(std::string_view path);
+
+  /**
+   * Construct a new AprilTagFieldLayout from a vector of AprilTag objects.
+   */
+  explicit AprilTagFieldLayout(const std::vector<AprilTag>& apriltags);
+
+  /**
+   * Set the alliance that your team is on.
+   *
+   * This changes the AprilTagFieldLayout::getTag(int) method to return the
+   * correct pose for your alliance.
+   *
+   * @param alliance the alliance to mirror poses for
+   */
+  void SetAlliance(DriverStation::Alliance alliance);
+
+  /**
+   * Gets an AprilTag pose by its id.
+   *
+   * @param id the id of the tag
+   * @return The pose corresponding to the id that was passed in.
```

Comment:
ID should be capitalized. We're not asking the user to pass the tag's https://en.wikipedia.org/wiki/Id,_ego_and_super-ego.